plugins/eslint/src/tasks/eslint.ts

```diff
@@ -1,13 +1,13 @@
 import { ToolKitError } from '@dotcom-tool-kit/error'
 import { styles } from '@dotcom-tool-kit/logger'
-import { Task } from '@dotcom-tool-kit/base'
+import { Task, TaskRunContext } from '@dotcom-tool-kit/base'
 import { ESLintSchema } from '@dotcom-tool-kit/schemas/lib/tasks/eslint'
 import { ESLint } from 'eslint'
 
 export default class Eslint extends Task<{ task: typeof ESLintSchema }> {
   static description = ''
 
-  async run(files?: string[]): Promise<void> {
+  async run({ files }: TaskRunContext): Promise<void> {
```

Comment:
**question(for @ivomurrell)**: typescript doesn't infer the type of this argument, do you know why?

Reply:
> It’s important to understand that an `implements` clause is only a check that the class can be treated as the interface type. It doesn’t change the type of the class or its methods _at all_.

https://www.typescriptlang.org/docs/handbook/2/classes.html#cautions

Guess you can imagine that the check that a subclasses properly implements a superclass comes after the subclass's types have been resolved.